app/src/main/assets/www/maps/patientsHomeLocationMap.html

```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <style>
+        /* Set the size of the div element that contains the map */
+        #map {
+            width:100% !important;
+            height:100% !important;
+            top:0px !important;
+            left:0px !important;
+            position: fixed !important;
+        }
+    </style>
+</head>
+<body>
+<div id="map"></div>
+<script>
+    //inject script
+
+    var Windows = [];
+    function bindInfoWindow(marker, map, infowindow, patientUuid) {
+        marker.addListener('click', function() {
+            closeOtherWindows();
+            infowindow.content; // 'content' is an element of infowindow
+            infowindow.open(map, this);
+            Windows[0] = infowindow;
+            patientsLocationMapInterface.setSelectedPatientUuid(patientUuid);
+            patientsLocationMapInterface.showGetDirectionsButton();
+        });
+
+        infowindow.addListener('closeclick', function () {
+            patientsLocationMapInterface.clearSelectedPatientUuid();
+            patientsLocationMapInterface.hideGetDirectionsButton();
+            infowindow.close();
+        });
+    }
+
+    function closeOtherWindows() {
+        if (Windows.length > 0) {
+            Windows[0].close();
+            Windows.length = 0;
+        }
+    }
+
+    function navigateToPatientSummary(){
+        patientsLocationMapInterface.navigateToPatientSummary();
+    }
+
+    function initMap() {
+        var homeDetails = patientsLocationMapInterface.getPatientsHomeDetailsForMapping();
+        if(homeDetails != null){
+            homeDetails = JSON.parse(homeDetails);
+        }
+
+        var currentLocation = patientsLocationMapInterface.getCurrentGPSLocation();
+        if(currentLocation != null){
+            currentLocation = JSON.parse(currentLocation);
+        }
+
+        var mapCenterLocation;
+
+        if(currentLocation != undefined && currentLocation.latitude != undefined && currentLocation.longitude != undefined){
+            mapCenterLocation = {lat: parseFloat(currentLocation.latitude) , lng: parseFloat(currentLocation.longitude)};
+        } else {
+
+            var latitudeTotals = 0;
+            var longitudeTotals = 0;
+            var count = 0;
+
+            for (var k in homeDetails) {
+                if (homeDetails[k].latitude != undefined && homeDetails[k].longitude != undefined) {
+                    latitudeTotals += parseFloat(homeDetails[k].latitude);
+                    longitudeTotals += parseFloat(homeDetails[k].longitude);
+                    count++;
+                }
+            }
+            if (count > 0) {
+                mapCenterLocation = {lat: latitudeTotals / count, lng: longitudeTotals / count};
+            } else {
+                mapCenterLocation = {lat: 0.5117, lng: 35.282614};
```

Comment:
To display a map for all clients, center map around current device location. If not available, calculate center location for all available patient address coordinates. If still not available, use default location (Eldoret - mUzima home) to avoid blank map, and show message to explain that no client location has been set.